docs/topics/addons.rst

```diff
@@ -0,0 +1,348 @@
+    (e.g. for external Python libraries) and update the
+    :class:`~scrapy.settings.Settings` object as wished, e.g. enable components
+    for this add-on or set required configuration of other extensions.
+
+    :param config: Configuration of this add-on
+    :type config: ``dict``
+
+    :param settings: The settings object storing Scrapy/component configuration
+    :type settings: :class:`~scrapy.settings.Settings`
+
+.. method:: check_configuration(config, crawler)
+
+    This method is called when the :class:`~scrapy.crawler.Crawler` has been
+    fully initialized, immediately before it starts crawling. You can perform
+    additional dependency and configuration checks here.
+
+    :param config: Configuration of this add-on
+    :type config: ``dict``
+
+    :param crawler: Fully initialized Scrapy crawler
+    :type crawler: :class:`~scrapy.crawler.Crawler`
+
+.. method:: update_addons(config, addons)
+
+    This method is called immediately before :meth:`update_settings`, and should
+    be used to enable and configure other *add-ons* only.
+
+    Add-ons that are added to the :class:`~scrapy.addons.AddonManager` during
+    this callback will also have their :meth:`update_addons` method called.
+
+    :param config: Configuration of this add-on
+    :type config: ``dict``
+
+    :param addons: Add-on manager holding all loaded add-ons
+    :type addons: :class:`~scrapy.addons.AddonManager`
+
+Additionally, add-ons may (and should, where appropriate) provide one or more
+attributes that can be used for limited automated detection of possible
+dependency clashes:
+
+.. attribute:: requires
+
+    list of built-in or custom components needed by this add-on, as strings.
+
+.. attribute:: modifies
+
+    list of built-in or custom components whose functionality is affected or
+    replaced by this add-on (a custom HTTP cache should list ``httpcache`` here)
+
+.. attribute:: provides
+
+    list of components provided by this add-on (e.g. ``mongodb`` for an
+    extension that provides generic read/write access to a MongoDB database)
+
+The entries in the :attr:`requires` and :attr:`modifies` attributes can be add-on
+names or components from other add-ons' :attr:`provides` attribute. You can
+specify :pep:`440`-style information about required versions. Examples::
+
+    requires = ['httpcache']
+    requires = ['otheraddon >= 2.0', 'yetanotheraddon']
+
+The Python object or module that is pointed to by an add-on path (e.g. given in
+the ``ADDONS`` setting, or given to
+:meth:`~scrapy.addons.AddonManager.add`) does not necessarily have to be an
+add-on. Instead, it can provide an ``_addon`` attribute. This attribute can be
+either an add-on or another add-on path.
+
+
+Add-on base class
+=================
+
+Scrapy comes with a built-in base class for add-ons which provides some
+convenience functionality:
+
+* basic settings can be exported via :meth:`~scrapy.addons.Addon.export_basics`,
```

Comment:
What does "exported" mean here?

Reply:
Basically you set a dict attribute that will be merged into Scrapy settings